### PR TITLE
docs: add cvss info and suggest further procedural details

### PR DESCRIPTION
**Pull request by pshiu**
Add information how the SWG handles:
- Security scoring
- Vulnerabilities for operators
- Follow up on remediation & on-call transition
- Disagreements on severity classification (and associated SLA)
- Temporary private forks
- Confidentiality
- Disclosure timeline
- Annual security survey

### CODE REVIEW

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 1]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+
+Maintainers will be reminded to remediate disclosures in proportion to the severity of the disclosure. The following table shows the reminder frequency until resolution of each severity classification:
+
+========    =====   ==================
+Severity    Score   Reminder frequency
+========    =====   ==================
+Low         ≥0.1    Once a semester
+Medium      ≥4.0    Once a quarter
+High        ≥7.0    Once a month
+Critical    ≥9.0    Once a week
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:96]: At first glance this seemed to infrequent but given that after the initial notification it's the maintainer's responsibility to prioritize and resolve the issue, I think it makes sense.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 2]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+To assist the maintainer with keeping track of their repository's vulnerabilities, the security working group will create a `GitHub repository security advisory`_ pre-assigned with the scored severity.
+
+Maintainers are requested to notify the security working group if they would like to adjudicate a vulnerability to a different severity than what was originally triaged. The security working group must accept the maintainer's adjudication, but should comment on any considerations around the adjudication.
+
+Maintainers will be reminded to remediate disclosures in proportion to the severity of the disclosure. The following table shows the reminder frequency until resolution of each severity classification:
+
+========    =====   ==================
+Severity    Score   Reminder frequency
+========    =====   ==================
+Low         ≥0.1    Once a semester
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:93]: What does semester mean in this context?  Is it meant to convey "twice annually"?

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 3]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+* Forward operator-specific vulnerabilities to any relevant Open edX operators
+* Score the severity of the disclosure
+* Identify the affected repositories
+* Open a GitHub repository security advisory for the vulnerability in the relevant repositories
+* Notify the Open edX maintainers of those repositories of the vulnerability
 
-  * Triage and calculate severity of issue
-  * Notify and work with SME for the impacted area(s)
-  * Get ticket prioritized and resolved (i.e. project manage the ticket(s) in SME backlogs)
+The security working group gives security severity and remediation advice to maintainers
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:68]:
```suggestion
Issue Triage and Routing
```
These section titles are very verbose, here's one suggestion for a shorter one but if you have a different idea, feel free to use that also.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 4]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
 
-Because the on-call member is expected to follow a disclosure through its eventual resolution, some of these
-responsibilities will extend beyond the initial scheduled on-call time.
+The security working group will also score incoming vulnerabilities using the `Common Vulnerability Scoring System (CVSS), version 3.1`_. This system provides a way to score vulernabilities from 0.0 to 10.0, which map to a severity of Low, Medium, High, or Critical.
+
+This numerical score and qualitative severity rating will be provided to maintainers so they are better informed on how severe the vulnerability is and have a sense of how important it may be to interrupt their work on the repository to resolve the vulnerability.
+
+.. _Common Vulnerability Scoring System (CVSS), version 3.1: https://www.first.org/cvss/v3.1/specification-document
+
+Maintainers are ultimately responsible for the security of their repositories
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:80]:
```suggestion
Maintainer Responsibilty
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 5]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+-----------------------------------------------------------------------------
+Maintainers are responsible for resolving disclosed security vulnerabilies and notifying the security working group once they are resolved.
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:82]:
```suggestion
Maintainers are ultimately responsible for resolving disclosed security vulnerabilities and notifying the security working group once they are resolved.
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 6]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
 
 .. note::
 
-   We would also like to specifically note that if an on-call week is particularly time intensive, the on-call member can
-   ask the other working group members for support.
+    Maintainers are encouraged to reach out to security@openedx.org if they need any help remediating security issues.
+
+.. _GitHub repository security advisory: https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/creating-a-repository-security-advisory
+
+Details of security vulnerabilities are private until a fix is released
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:107]:
```suggestion
Security Vulnerability Privacy
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 7]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+-----------------------------------------------------------------------
+Members of the Open edX community with access to the details of a disclosure are asked to keep those details private until a fix is publically released for the vulnerability.
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:109]:
```suggestion
Details of security vulnerabilities will remain private until a fix is released. Members of the Open edX community with access to the details of a disclosure are asked to keep those details private until a fix is publicly released for the vulnerability.
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 8]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+
+Maintainers should use GitHub's `temporary private fork`_ feature within the GitHub repository security advisory created by the security working group to keep the implementation details of a fix for a vulnerability private.
+
+.. _temporary private fork: https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/collaborating-in-a-temporary-private-fork-to-resolve-a-repository-security-vulnerability
+
+Operators are ultimately responsible for the security of their Open edX instance
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:115]: Any reason this shouldn't just go into the `Guidance for operators` section?

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 9]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+--------------------------------------------------------------------------------
+The security working group will put forth good faith effort to responsibly forward disclosures related to the operation of a specific Open edX instance to the relevant operator.
+
+Operators wishing to register a desired primary point of contact for security-related matters should email security@openedx.org with the name, domain, contact information for an individual point of contact for security, and an email address to which security reports should be forwarded.
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:119]:
```suggestion
Operators wishing to register a desired primary point of contact for security-related matters should email security@openedx.org with the name, domain, contact information for an individual point of contact for security, and an email address to which security reports should be forwarded.  We recommend that you provide a consistent alias for this rather than the e-mail address of a specific person to keep the process resilient.  e.g. ``security@companyx.com``
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 10]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+
+Operators are encouraged to forward or to invite reporters to forward to security@openedx.org any security issue they receive that may be relevant to the Open edX project.
 
 .. _security-releases:
 
-Security Releases
------------------
+Maintainers will work together with the on-call security working group member to release the security fix
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:125]: Why this change, the shorter title seems like it captures what this section is about.  Maybe something in between like `Vulnerability Release Process`

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 11]
```diff
@@ -51,52 +51,96 @@ Reports come into security@openedx.org
...
+---------------------------------------------------------------------------------------------------------
 The current process for releasing security fixes involves sending a disclosure and security patch to members of the
 open source security email list and waiting two days before making the patch public.  Instead of this process, we
 propose the following security release process:
 
-#. A notice will be sent to everyone to announce an upcoming security patch, specifying the date and time at which
-   the patch will become public
+#. The maintainer will email security@openedx.org notifying that a fix is ready to be merged and the date and time of when they would like to merge it. The merge time should be at least 72 hours after the security patch is announced.
+
+#. The on-call security working group member will create an announcement post in the `Security Announcement Section`_ on https://discuss.openedx.org. It should specify the affected repository, the date and time at which the patch will become public, and the severity of the vulnerability it fixes.
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:133]: Gut reaction: This feels like something we could just ask the maintainer to do if we provide them a doc to follow, but I'm fine with it being a SWG responsibility if you feel strongly about it.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 12]
```diff
@@ -110,14 +154,17 @@ Examples
...
 
 Consequences
 ************
 
+
 Security Working Group
 ======================
 
-Invite Only
+Invite only
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:167]:
```suggestion
Invite Only
```
Shouldn't titles be fully capitalized?  Using [this](https://www.grammarly.com/blog/capitalization-in-the-titles/) as my guide.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 13]
```diff
@@ -130,24 +177,29 @@ Proactive work for security improvements will be made public, while vulnerabilit
 We propose using Github Security Advisories to handle the triage of vulnerability reports to the owners of
 vulnerable components.
 
-Security Team Member Responsibilities
--------------------------------------
+Member responsibilities
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:180]:
```suggestion
Member Responsibilities
```

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 14]
```diff
@@ -156,6 +208,15 @@ What do I do if I am an operator and someone reports a vulnerability to me?
...
+What will happen if a report is accidentally sent to security@openedx.org for the operation of my Open edX instance?
+
+* Please let security@openedx.org know the best email to forward such reports to, along with your Open edX instance name, domain, and separate contact information for an indidual responsible for security at your organization. The security working group will do its best to forward such reports to the correct organization.
+
+How do I receive notification of the release of upcoming security patches?
+
+* Please watch the Open edX Discourse Security Announcements topic at https://discuss.openedx.org/c/announcements/security/19. If you are logged in, the button with a bell icon on the top right corner above the topic list will let you customize how you are notified of security announcements.
+
+
 Change History
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:220]: Don't forget to add a section to the change history.

**Review by feanil — COMMENTED** [review]
This are really great additions (please don't let my heavy suggesting make you think otherwise!)

Most of my change suggestions are stylistic so let me know if it would be easier to just get on a video call to go through them together and arrive at something we're both happy with.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 2 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:93]: Yes! I will amend.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 8 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:115]: You're right, most of the information is down there already. Removed.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 3 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Great suggestion.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 4 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:80]: Will commit.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 5 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Will commit.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 6 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:107]: Will commit.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 7 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Will commit.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 9 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:119]: Since I deleted this section per your previous comment, I moved the spirit of this change to the Guidance for Operators section. Let me know what you think and please feel free to propose other edits.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 10 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Was trying to keep the parallelism with the long form of the first section title! Will rename back to `Security Releases`.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 11 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Not at all, I prefer less work honestly. Amended.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 12 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:167]: Yes, was imitating the first title `Reports come into security@openedx.org`. Amended.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 15]
```diff
@@ -124,30 +166,37 @@ volunteers are needed, we will put out a call to join. Volunteers will need to a
 and will then be evaluated by the existing working group members. We hope to have six to seven members of the working
 group at a time.
 
-Security work will be a mix of private and public tickets
----------------------------------------------------------
-Proactive work for security improvements will be made public, while vulnerability reports will be handled privately.
-We propose using Github Security Advisories to handle the triage of vulnerability reports to the owners of
```
Comment (pshiu) [inline oeps/processes/oep-0060-proc-sec-group.rst:130]: @feanil I think for GitHub security advisories to work for us, we will need to grant maintainers repository ownership / GitHub repository admin permissions on this repo. Is this in alignment with tCRIL's plans for maintainership?

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 14 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:217]: Done.

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 15 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:130]: (What I am reading: https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/permission-levels-for-repository-security-advisories)

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 16]
```diff
@@ -171,6 +171,8 @@ volunteers are needed, we will put out a call to join. Volunteers will need to a
 and will then be evaluated by the existing working group members. We hope to have six to seven members of the working
 group at a time.
 
+There will also be ways to participate in the security of the Open edX platform without being a member of the working group. Maintainers and core contributors of the Open edX community interested in volunteering to implement security hotfixes for other maintainers who may not have the bandwidth to immediately address vulnerabilities are encouraged to email their interest to security@openedx.org.
```
Comment (pshiu) [inline oeps/processes/oep-0060-proc-sec-group.rst:169]: @alangsto @feanil Wanted to draw your attention to this line that I added post-PR-creation to make sure you saw it and provide your thoughts!

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 15 above)
Comment (feanil) [reply oeps/processes/oep-0060-proc-sec-group.rst:130]: @pshiu I think there are a couple of ways to solve this but at the very least I expect the security working group will need admin access to all the open source repos to do their job.  I think we're okay with this.

For the larger question of maintainers and admin,  I'm gonna think about it more but it's come up in a few other places as well so I think we'll probably add an ADR to the maintainers OEP soon.

File: oeps/processes/oep-0060-proc-sec-group.rst [hunk 17]
```diff
@@ -124,30 +166,37 @@ volunteers are needed, we will put out a call to join. Volunteers will need to a
...
-vulnerable components.
+There will also be ways to participate in the security of the Open edX platform without being a member of the working group. Maintainers and core contributors of the Open edX community interested in volunteering to implement security hotfixes for other maintainers who may not have the bandwidth to immediately address vulnerabilities are encouraged to email their interest to security@openedx.org.
+
+Visibility
+----------
+Security work will be a mix of private and public tickets. Proactive work for security improvements will be made public, while vulnerability reports will be handled privately.
 
-Security Team Member Responsibilities
--------------------------------------
+We propose using GitHub Security Advisories to handle the triage of vulnerability reports to the owners of vulnerable components.
```
Comment (feanil) [inline oeps/processes/oep-0060-proc-sec-group.rst:175]: Not a problem to note here but I'm realizing that we'll need to keep a master list of draft advisories as a working group because github doesn't really provide a way to do that yet at the organization level.

**Review by feanil — APPROVED** [review]

File: oeps/processes/oep-0060-proc-sec-group.rst (hunk 15 above)
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst:130]: Always happy with more permissions, but there is a beta Security Manager role GitHub offers too that we might be able to tinker around with. [[docs](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/managing-security-managers-in-your-organization)]
